src/local/room_xy/approximate_offsets.rs

```diff
@@ -0,0 +1,236 @@
+//! Methods related to approximating in-room positions
+//! between other in-room positions.
+
+use super::RoomXY;
+
+impl RoomXY {
+    /// Calculates an approximate midpoint between this point and the target.
+    ///
+    /// In case of a tie, rounds towards this point.
+    ///
+    /// If `distance_towards_target` is bigger than the distance to the target,
+    /// the target is returned.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use screeps::RoomXY;
+    ///
+    /// // Exact distances
+    /// let start = RoomXY::checked_new(10, 10).unwrap();
+    /// let target = RoomXY::checked_new(10, 15).unwrap();
+    /// assert_eq!(
+    ///     start.towards(target, 1),
+    ///     RoomXY::checked_new(10, 11).unwrap()
+    /// );
+    /// assert_eq!(
+    ///     start.towards(target, 4),
+    ///     RoomXY::checked_new(10, 14).unwrap()
+    /// );
+    /// assert_eq!(
+    ///     start.towards(target, 10),
+    ///     RoomXY::checked_new(10, 15).unwrap()
+    /// );
+    ///
+    /// // Approximate/rounded distances
+    /// let start = RoomXY::checked_new(10, 10).unwrap();
+    /// let target_1 = RoomXY::checked_new(15, 20).unwrap();
+    /// let target_2 = RoomXY::checked_new(0, 5).unwrap();
+    /// assert_eq!(
+    ///     start.towards(target_1, 1),
+    ///     RoomXY::checked_new(10, 11).unwrap()
+    /// );
+    /// assert_eq!(
+    ///     start.towards(target_1, 9),
+    ///     RoomXY::checked_new(14, 19).unwrap()
+    /// );
+    /// assert_eq!(
+    ///     start.towards(target_2, 1),
+    ///     RoomXY::checked_new(9, 10).unwrap()
+    /// );
+    /// ```
+    pub fn towards(self, target: RoomXY, distance_towards_target: i8) -> RoomXY {
```

Comment:
This is a bit of a weird one; wouldn't you want a float in `[0,1]` for `self*x + target*(1-x)`?

Reply:
I'm not entirely sure what you're asking here. Can you clarify a bit more?

Reply:
Ah, yes, this is a directional interpolation method. I'm not sure that returning the interpolation multiplier (or even vector) is all that helpful, though, since the user would still need to determine what `RoomXY` corresponds to that interpolation multiplier.

As with `PartialOrd`, this was copied from `Position` for API parity.